src/colmap/scene/reconstruction.cc

```diff
@@ -125,11 +125,11 @@ void Reconstruction::AddImage(class Image image) {
     image.SetCameraPtr(&camera);
   }
   const image_t image_id = image.ImageId();
-  const bool is_registered = image.IsRegistered();
+  const bool is_registered = image.HasPose();
```

Comment:
same here

Reply:
No, here we specifically check whether the new image should be set as registered.